### PR TITLE
Modified html to match github's default styling

### PR DESCRIPTION
**Pull request by xt0rted**
This was brought up in https://github.com/StylishThemes/GitHub-Dark/issues/1088. Not all scenarios were tested (including GitLab) so there might be some tweaking needed.

The overall html is based on the [Dependency Graph](https://github.com/npmhub/npmhub/network/dependencies) page and the utility classes in [Primer](https://primer.style/css/utilities).

Preview of what the changes look like.

image

### CODE REVIEW

File: source/content.css [hunk 1]
```diff
@@ -3,13 +3,6 @@
...
   margin: 0 !important; /* gitlab */
 }
-.npmhub-deps > li {
-  padding: 10px;
-  border-bottom: 1px solid #DDD;
-}
-.npmhub-deps > li:last-child {
-  border-bottom: none;
-}
 li.npmhub-empty {
```
Comment (xt0rted) [inline source/content.css]:
```suggestion
.npmhub-empty {
```

**Comment by fregante** [comment]
Excellent, thank you! I'll test it in the following days

**Comment by the-j0k3r** [comment]
> I'm not sure what it's like with GitLab though.

Yea that PR heavily uses GitHubs specific classes, what if npmhub sourced such CSS separately so a stylesheet for GitLab or GitHub instead of making it purely GitHub specific.

Not sure if these classes will be simply dropped in when using the extension in GitLab only or not though. So the tests will be interesting.

**Comment by fregante** [comment]
The spacing margin is missing:

spacing

And it feels a little tight:

<table>
<tr>
<th> This PR
<th> Files table
<th> `master`
<tr>
<td>
pr

<td>
files

<td>
master

</table>

Any way to restore the spacing? Maybe just `p-1` classes on each line. I think that given how "heavy" and full these lines are, they _need_ more space than files.

**Comment by the-j0k3r** [comment]
Is this really how that looks?

77706146-d91ab980-6fc1-11ea-839b-c30f20e0efc5

The spacing between these seems not OK in this screenshot.

**Comment by fregante** [comment]
Yep

**Comment by xt0rted** [comment]
I adjusted the html some more and this is what it looks like now in Windows 10.

Chrome 80.0.3987.149 | FireFox 74.0b9
-- | --
image | image

Do you have a link to a GitLab page I can test this on?

**Comment by the-j0k3r** [comment]
> Do you have a link to a GitLab page I can test this on?

https://gitlab.com/RaitaroH/Stylus-DeepDark for instance?

**Comment by xt0rted** [comment]
Turns out if I read the source there's a sample url in it 😄

If I go to https://gitlab.com/gitlab-org/gitlab-foss/blob/master/package.json the dependencies load, but not on https://gitlab.com/RaitaroH/Stylus-DeepDark/-/blob/master/package.json 🤷‍♂

After my updates everything looks good on GitHub but it's pretty much unstyled on GitLab. @fregante is that something you'd like to do in a separate PR?

File: source/content.js [hunk 2]
```diff
@@ -31,8 +31,12 @@ function isPackageJson() {
 }
 
 function addHeaderLink(box, name, url) {
+  isGitLab() ?
   box.firstElementChild.prepend(doma(`
     <a class="btn btn-sm BtnGroup-item" href="${url}">${name}</a>
+  `)) :
+  box.querySelector('.Box-header').append(doma(`
+    <a class="btn btn-sm" href="${url}">${name}</a>
```
Comment (fregante) [inline source/content.js]: `BtnGroup-item` is required or else the buttons appear as:

split

I'd avoid this ternary. `querySelector` can work in both cases, with the right selector

File: source/content.js [hunk 3]
```diff
@@ -112,14 +116,16 @@ async function getPackageJson() {
...
-    <div class="Box Box--condensed mt-5 file-holder">
-      <div class="npmhub-header BtnGroup"></div>
+    <div class="Box mt-5">
       ${
         isGitLab() ?
-        `<div class="file-title"><strong>${title}</strong></div>` :
-        `<h3 class="Box-header Box-title px-2">${title}</h3>`
+        `<div class="npmhub-header BtnGroup"></div>
+        <div class="file-title"><strong>${title}</strong></div>` :
+        `<div class="Box-header lh-condensed py-2 d-flex flex-items-center">
```
Comment (fregante) [inline source/content.js]: `BtnGroup` is a GitHub class and is needed for the same reason as above

**Comment by fregante** [comment]
GitLab's borders selector is: `.tree-holder .tree-table tr`.

I don't see GitHub-like utility classes to apply borders on GitLab only either.

Maybe it's easier to just make it a regular `table`

**Comment by fregante** [comment]
Thank you for opening the PR @xt0rted. I just merged #105 which makes it easier to reason about the HTML, but sadly is incompatible with this PR.

If anyone wants to retry this: drop all the CSS except the `pulse` and `opacity` code, and just use native classes. This should be even easier now that GitLab isn't supported at all until #106 is fixed

**Pull request closed without merging**